contracts/Core/StateManager.sol

```diff
@@ -1,17 +0,0 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.0;
-
-import "../lib/Constants.sol";
-
-
-contract StateManager {
```

Comment:
I think we should keep State Manager as it was before. There are two reasons. First :  Constants contract should only have constants which are static. so its clear around, Second : Architecture remains readable. I am not sure what is benefit of keeping it in constants ? @dev1644 Please let me know if there are any ?

Reply:
> I think we should keep State Manager as it was before

`StateManager` only had `getEpoch` & `getState` functions, these functions heavily relied upon data from particular values (that are provided by Constants lib/contract). 

Every external calls from a contract to another has some fixed stipend to them (excluding computation cost of that function).
Since only functionality of `StateManager` was just to compute & fetch current Epoch & State (which in turn relies upon values provided by Constants lib/contract), it makes sense to have Constants contract/library to encapsulate `getEpoch & getState` functionality as well in a single contract (`StateManager` offers nothing to have it in its own external contract, at least for now) from gas expenses & complexity perspective.

> I am not sure what is benefit of keeping it in constants ?

Benefits of keeping it in constants is to save few thousand gas (Need more analysis to calculate exact amount).

Reply:
Yes, my bad ! lower gas consumption should be our top-most priority, naming, architecture comes later
Okay 
Can we do it like this maybe
Define those two constants: numStates and epochLength in State Manager only
as numStates is only being used my State Manager 
whereas though epochLength is being used by BlockManager as well, there we have an instance of State Manager, so we can access it if we define epochLength in State Manager

**Please check I think this would solve my arguments and yours as well.** 

State Manager  - All parameters related to state
> numStates
> epochLength
> commit id (adding these here would make more sense )
> reveal id
> propose id
> dispute id

But again one can argue that we want all constants in one place :)
To solve this 
From constants, we can have an external call to StateManager to set up these values, 
So our governing remains in constants.sol only
 
What do you think ?

Reply:
> Define those two constants: numstates and epochLength in State Manager only 

Separating constants (just referring to them as constants, it's clear these params are not constants) into a different contract doesn't make sense to me, some of the constants will be in X contract & some of them in Y contract. Moreover, the current implementation of `StateManager` is more of a `StateViewer/Getter` not of a `Manager`.

> Define those two constants: numStates and epochLength in State Manager only as numStates is only being used my State Manager anyway

Yes, `numStates` is being used in `StateManager` only, in `getState` function but in turn, this function is being used much time in multiple contracts (`BlockManager`, `VoteManager`, `StakeManager`).

I understand that some of the parameters in constants lib/contract are never gonna change (this need to be confirmed again though), once confirmed I think it would be better if these immutable constants parameters will exist in all of the contracts with the `constant` type (it will increase one time-deployment gas but will save runtime gas during tx).  This needs to discussed first.

>  Have set for epochLength as well from here by external call

Again, I am not seeing the incentive for using two external calls to set/get a parameter when it can be done with one call only. 


Regarding `never changing constants parameters`, I think it's better to dissolve `Structs` lib too & have a common `Storage` contract which will be inherited by another contract, `never changing constants parameters` will exists in `Storage` contract as of `constant type`.

Reply:
Okay looks good to me as of now
Let's Merge this now 
And we will create issues with milestone v2 for open points CC: @hrishikeshio @SkandaBhat @SamAg19 
Can you please create @dev1644?

**Issue 1**
Define never changing constants in contracts only to avoid runtime external calls.

**Issue 2**
 Actually, I am not totally clear on following, Please explain it issue

> Regarding never changing constants parameters, I think it's better to dissolve Structs lib too & have a common Storage contract which will be inherited by another contract, never changing constants parameters will exists in Storage contract as of constant type.

**Issue3**
Replace Admin role with governer role in constants.sol

Reply:
Issue 1 and 2 already exist. We need to create issues for them and track.